RecoEcal/EgammaClusterAlgos/src/PFECALSuperClusterAlgo.cc

```diff
@@ -392,6 +392,13 @@ buildSuperCluster(CalibClusterPtr& seed,
 	<< std::endl;
     }    
   }
+
+  //temporary fix for HGCal issue
+  if (not_clustered == clusters.begin()) {
```

Comment:
IIUC, the problem affects cases of this code running on HGCAL inputs.
is it obvious to give no effect on regular ECAL clustering?
If not, please add a check that this is HGCAL.

Reply:
If this ever happened in ECAL, we would have noticed the algorithm crashing before now.

My understanding from @rovere's investigation is that this can only happen for HGCal because of the different meaning of calibrated vs uncalibrated energy.

Reply:
@lgray  [as the main developer of this in the past]
please check if this change is reasonable

Reply:
So, if I understand correctly, the buildSuperCluster is supposed to be called only on seedable clusters.
The proposed change is a quiet escape from a bug introduced by the caller where the selection of seedable clusters contradicts any ability to match the list with a seed.
This is not good.

The solution appears to be OK for the current use cases of buildSuperCluster, but it opens up the property of this method to handle unseedable list of clusters, which should not be exercised.
I propose to add a config parameter (e.g. ```dropUnseedable ``` set to false by default ) and set it to true for HGCAL and also add a LogError or even an exception if dropUnseedable is false and we end up here.

Reply:
Ciao @slava77 the real fix in the realistic simcluster energy assignment will automatically fix this issue and will not crash. Having said that, the code as is is still fragile since, as I explained in the text reported above by @kpedro88 the conditions to be seedable are not safe for the mathematical operations that will follow in doing the partitioning of the clusters.
So, if you prefer, the code is not protected but feeding it with decent cluster won't crash.
Whether we want to consider it as a bug or a feature is, in a sense, somewhat personal.